### DISCUSSION TITLE
Executing Azure Devops backlog tasks via Airflow DAGs

### DISCUSSION THREAD

**Discussion by SajalRoy** [apache/airflow#25868] [Q&A] [answered]
Keen on using Airflow DAGs for executing devops backlog tasks ( which are essentially data report generation pipelines) . For eg. User has raised a backlog task to run certain DAG with all config information available in the task. Can Airflow pick it up from the backlog list and execute it as a Normal DAG with the config information as given in the tasks.

**Comment by potiuk** [comment, marked as answer]
Yes and no. So far, Airflow is based on Scheduling. You can also trigger Dag Runs using API, so if you can write some trigger to run a DAG after such backlog is updated (Push mode), Airflow can process such backlog.

Howver  in the upcoming 2.4 version (few weeks/2 months likely) we are going to have a new feature by implementing AIP-48 : https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-48+Data+Dependency+Management+and+Data+Driven+Scheduling  - which will allow Airlfow to trigger DAGs based on new data appearing (so in short you will be able to trigger DAGs in pull mode without having to trigger DAG via API).

**Reply by SajalRoy** [reply]
Thanks @potiuk ! This helps alot . Awaiting the new 2.4 version :)
